x/market/client/cli/flags.go

```diff
@@ -80,6 +80,20 @@ func BidIDFromFlagsWithoutCtx(flags *pflag.FlagSet) (types.BidID, error) {
 	return types.MakeBidID(prev, addr), nil
 }
 
+// ProviderFromFlagsWithoutCtx returns Provider address with given flags and error if occurred
+func ProviderFromFlagsWithoutCtx(flags *pflag.FlagSet) (sdk.AccAddress, error) {
```

Comment:
add & use the accompanying `AddProviderFlags()` - this makes it so that all the commands use consistent flag names.